drivers/spi/spi_mchp_mec5_qspi.c

```diff
@@ -0,0 +1,542 @@
+/*
+ * Copyright (c) 2025 Microchip Technology Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT microchip_mec5_qspi
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_mec5, CONFIG_SPI_LOG_LEVEL);
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/dt-bindings/spi/spi.h>
+#include <zephyr/sys/sys_io.h>
+#include <zephyr/sys/util.h>
+#include <soc.h>
+#include <zephyr/irq.h>
+
+#include "spi_context.h"
+
+/* MEC5 HAL */
+#include <device_mec5.h>
+#include <mec_ecia_api.h>
+#include <mec_espi_taf.h>
+#include <mec_qspi_api.h>
+
+struct mec5_spi_devices {
+	uint32_t cs_timing;
+	uint8_t cs;
+	uint8_t sck_tap;
+	uint8_t ctrl_tap;
+	uint8_t flags;
+};
+
+/* Device constant configuration parameters */
+struct mec5_qspi_config {
+	struct mec_qspi_regs *regs;
+	int clock_freq;
+	const struct pinctrl_dev_config *pcfg;
+	void (*irq_config_func)(void);
+	uint8_t ovrc;
+	uint8_t num_child_devices;
```

Comment:
these 2 attributes could go at the end for better sturct alignment

Reply:
Hi @tbursztyka thanks for the review.  Made the recommended changed and pushed the update.